Add `with torch.no_grad()` to DistilBERT integration test forward pass

jaketae opened the pull request:
# What does this PR do?

This PR encapsulates forward passes in DistilBERT unit tests with `with torch.no_grad():` as per #14642.

Also, a unit test seemed to terminate early due to a `return` in a loop; this was replaced with `continue`. (If this was not what was intended, I will rollback the modification.)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?

@NielsRogge @LysandreJik @sgugger

jaketae committed:
refactor: wrap forward pass around no_grad context

jaketae commented on tests/test_modeling_distilbert.py:
```diff
@@ -261,7 +266,7 @@ def test_torchscript_device_change(self):
 
             # BertForMultipleChoice behaves incorrectly in JIT environments.
```
Do we want to skip JIT testing for `DistilBertForQuestionAnswering`, `DistilBertForSequenceClassification`, and `DistilBertForTokenClassification` as well (these classes come after `BertForMultipleChoice`).

sgugger reviewed (commented):
I'm not sure those checks were meant under the scope of the issue. I think it was only for the last integration test, but I'll let @LysandreJik confirm.

LysandreJik reviewed (commented):
The request was indeed meant only for the integration tests :)

jaketae commented on tests/test_modeling_distilbert.py:
```diff
@@ -261,7 +266,7 @@ def test_torchscript_device_change(self):
             # BertForMultipleChoice behaves incorrectly in JIT environments.
             if model_class == DistilBertForMultipleChoice:
-                return
+                continue
```
```suggestion
                return
```

jaketae committed:
Update tests/test_modeling_distilbert.py

jaketae commented:
Thank you for the feedback and confirmation! I've reverted the change.

LysandreJik commented:
Hello @jaketae, were your last changes committed? I still see a lot of non-integration tests with `torch.no_grad`

jaketae committed:
fix: rm `no_grad` from non-integration tests

jaketae committed:
chore: rm whitespace change

jaketae commented:
@LysandreJik Apologies for the delay! I thought I had properly committed and pushed the changes but seems like that wasn't the case. Reverted most of the `no_grad` context and modified only the integration test. Thank you!

sgugger reviewed (approved):
Thanks a lot!
